Gem/Code/Source/GameLift/GameLiftClientComponent.h

```diff
@@ -0,0 +1,98 @@
+
+        void HostSessionOnQueue(const AZ::ConsoleCommandContainer& consoleFunctionParameters);
+        AZ_CONSOLEFUNC(GameLiftClientSystemComponent, HostSessionOnQueue, AZ::ConsoleFunctorFlags::DontReplicate, "Host and join a new session on queue");
+
+        void JoinSession(const AZ::ConsoleCommandContainer& consoleFunctionParameters);
+        AZ_CONSOLEFUNC(GameLiftClientSystemComponent, JoinSession, AZ::ConsoleFunctorFlags::DontReplicate, "Join an existing session");
+
+        void SearchSessions(const AZ::ConsoleCommandContainer& consoleFunctionParameters);
+        AZ_CONSOLEFUNC(GameLiftClientSystemComponent, SearchSessions, AZ::ConsoleFunctorFlags::DontReplicate, "Search for existing sessions");
+
+        void LeaveSession(const AZ::ConsoleCommandContainer& consoleFunctionParameters);
+        AZ_CONSOLEFUNC(GameLiftClientSystemComponent, LeaveSession, AZ::ConsoleFunctorFlags::DontReplicate, "Leave the current session");
+
+        void StartMatchmaking(const AZ::ConsoleCommandContainer& consoleFunctionParameters);
+        AZ_CONSOLEFUNC(GameLiftClientSystemComponent, StartMatchmaking, AZ::ConsoleFunctorFlags::DontReplicate, "Start a matchmaking request");
+
+        void StopMatchmaking(const AZ::ConsoleCommandContainer& consoleFunctionParameters);
+        AZ_CONSOLEFUNC(GameLiftClientSystemComponent, StopMatchmaking, AZ::ConsoleFunctorFlags::DontReplicate, "Stop a matchmaking request");
+
+        ////////////////////////////////////////////////////////////////////////
+        // SessionAsyncRequestNotificationBus handler implementation
+        void OnCreateSessionAsyncComplete(const AZStd::string& createSessionReponse) override;
+
+        void OnSearchSessionsAsyncComplete(const AzFramework::SearchSessionsResponse& searchSessionsResponse) override;
+
+        void OnJoinSessionAsyncComplete(bool joinSessionsResponse) override;
+
+        void OnLeaveSessionAsyncComplete() override;
+
+        ////////////////////////////////////////////////////////////////////////
+        //MatchmakingAsyncRequestNotificationBus handler implementation
+        void OnAcceptMatchAsyncComplete() override;
+
+        void OnStartMatchmakingAsyncComplete(const AZStd::string& matchmakingTicketId) override;
+
+        void OnStopMatchmakingAsyncComplete() override;
+
+        ////////////////////////////////////////////////////////////////////////
+        //MatchmakingNotificationBus handler implementation
+        void OnMatchAcceptance() override;
+        void OnMatchComplete() override;
+        void OnMatchError() override;
+        void OnMatchFailure() override;
+
+         ////////////////////////////////////////////////////////////////////////
+
+    private:
+        void AcceptMatch(bool accept = true);
+        void JoinSessionInternal(const AZStd::string& sessionId, const AZStd::string& playerId);
+
+        AZStd::string m_ticketId;
+        AZStd::string m_playerId;

```

Comment:
nit: Add comment this is a locally generated player id, devs should feel free to replace this id with any other suitable id. It just uniquely identifies this player in a session to GameLift.